### PR TITLE
Issue_435:Convert Pytest-CVP L2 VLAN Name test case to Vane #435

### PR DESCRIPTION
**Pull request by rahulrajolegslab**
# Please include a summary of the changes

test_definition.yaml.j2: Updated J2 with testcase NRFU5.4
master_def.yaml: Updated master def with testcase NRFU5.4
test_definition.yaml: Added test def with testcase NRFU5.4
test_security_rp_l2_vlan_name.py: Added python file for testcase NRFU5.4

# Any specific logic/part of code you need extra attention on

Explain any specific logic you need special attention on
This testcase will verify all configured VLANs have a name configured.

# Include the Issue number and link

https://github.com/aristanetworks/vane/issues/435
Make sure to link the issue in the github PR UI 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU test case development)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

Unit test cases:

1. Verified that testcase pass if all configured VLANs have a name configured.
               
[all_vlan_name_configured.docx](https://github.com/aristanetworks/vane/files/12669378/all_vlan_name_configured.docx)
[report.zip](https://github.com/aristanetworks/vane/files/12669382/report.zip)
[j2 run pass.zip](https://github.com/aristanetworks/vane/files/12669385/j2.run.pass.zip)

2. Verified that testcase failed if any VLANs have not a name configured.

[no_name_vlan_configured.docx](https://github.com/aristanetworks/vane/files/12669389/no_name_vlan_configured.docx)
[report.zip](https://github.com/aristanetworks/vane/files/12669393/report.zip)
[j2 run fail.zip](https://github.com/aristanetworks/vane/files/12669400/j2.run.fail.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/test_definition.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+- name: nrfu_tests
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:1]: This test case should be added to sample_network_tests/nrfu

image

File: nrfu_tests/test_definition.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  testcases:
+    - name: test_security_rp_l2_vlan_name
+      description: Testcase for verification of security rp l2 vlan name functionality.
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:4]: Test cases for verification of configured VLAN naming.

File: nrfu_tests/test_definition.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,11 @@
+- name: nrfu_tests
+  testcases:
+    - name: test_security_rp_l2_vlan_name
+      description: Testcase for verification of security rp l2 vlan name functionality.
+      test_id: NRFU5.4
+      show_cmd: show vlan
+      expected_output: null
+      test_criteria: All configured VLANs should have a name configured.
+      report_style: modern
+      criteria: {{ global_dut_filter["criteria"] }}
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml.j2:10]:
```suggestion
      criteria: {{ global_dut_filter.criteria }}

```

File: nrfu_tests/test_security_rp_l2_vlan_name.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2022 Arista Networks, Inc.  All rights reserved.
+# Arista Networks, Inc. Confidential and Proprietary.
+
+"""
+Test cases for verification of security rp l2 vlan name functionality
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_l2_vlan_name.py:5]:
```suggestion
Test cases for verification of configured VLAN naming.

```

File: nrfu_tests/test_security_rp_l2_vlan_name.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+from vane.logger import logger
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_test
+@pytest.mark.security
+class SystemHardwareFreeMemoryTests:
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_l2_vlan_name.py:19]:
```suggestion
class VlanNameTests:

```

File: nrfu_tests/test_security_rp_l2_vlan_name.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            output = dut["output"][tops.show_cmd]["json"]
+            logger.info(
+                "On device %s, output of %s command is: \n%s\n",
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+
+            vlans = output.get("vlans")
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_l2_vlan_name.py:58]: Please add a assert here to confirm that vlans are present on device. Also you can filter out dynamic vlans here itself and then only check names are present on collected VLANS.

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

File: sample_network_tests/nrfu/test_security_rp_l2_vlan_name.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                for vlan, vlan_status in tops.expected_output.get("vlans").items():
+                    for vlan_name_status in vlan_status:
+                        expected_vlan_name = (
+                            tops.expected_output.get("vlans").get(vlan).get(vlan_name_status)
+                        )
+                        actual_vlan_name = (
+                            tops.actual_output.get("vlans").get(vlan).get(vlan_name_status)
+                        )
+                        if expected_vlan_name != actual_vlan_name:
+                            no_name_configured.append(vlan)
```
Comment (vishwajeetgslab) [inline sample_network_tests/nrfu/test_security_rp_l2_vlan_name.py:89]:
```suggestion
            if tops.actual_output != tops.expected_output:
                tops.output_msg = ""
                no_name_vlans = []
                for vlan, vlan_status in tops.expected_output.get("vlans").items():
                        actual_vlan_naming = (
                            tops.actual_output.get("vlans").get(vlan).get("configured_vlan_name")
                        )
                        if vlan_status["configured_vlan_name"] != actual_vlan_naming:
                            no_name_vlans.append(vlan)

```

File: sample_network_tests/nrfu/test_security_rp_l2_vlan_name.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                tops.output_msg += (
+                    f"Following VLANs are no name configured: {', '.join(no_name_configured)}."
```
Comment (vishwajeetgslab) [inline sample_network_tests/nrfu/test_security_rp_l2_vlan_name.py:91]: \nFor following VLANs name is not configured:

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

**Review by vishwajeetgslab — APPROVED** [review]

**Review by MaheshGSLAB — CHANGES_REQUESTED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Pull request merged**